Automattic/pocket-casts-ios
# Add "Change Avatar" button to account settings

Fixes # gravatarp2 -> 2024/06/10/pt-integrating-gravatar-into-pocket-casts

Adding the ability to change the avatar. Avatar is coming from user's Gravatar account. So the "Change Avatar" button opens the Gravatar web link via in-app Safari.

<img src="https://github.com/Automattic/pocket-casts-ios/assets/5032900/3b914610-ed16-451a-b62b-79f9a2380e9a" width=300/>


## To test

Profile > Account > "Change Avatar"
Observe: In app Safari is presented
   - Inside the browser, if you are not logged in you should see the login first, then you'll be redirected to avatar update page.
   - The avatar change does not always get reflected to the app immediately even though we force refresh the avatar on the mobile side skipping all client-side caches. This is a known limitation. There's a backend cache too and it's not updated immediately. The plan is to show such a message inside the browser:
   
 <img src="https://github.com/Automattic/pocket-casts-ios/assets/5032900/1eb34008-c439-4f3f-be2a-0501822ffb03" width=300/>

If you are not logged in to the app then "Change Avatar" button will not be present for you.

Known issue:

A strange auto-zoom happens when testing in a Simulator but this issue is not reproduced in real device. But let me know if you can reproduce this in a real device.

<img src="https://github.com/Automattic/pocket-casts-ios/assets/5032900/ddbd4148-0e7b-466c-9c6d-caa5e4545045" width=200/>

## Checklist
